models/project_board.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+type (
+	// ProjectBoardType is used to represent a project board type
+	ProjectBoardType uint8
+
+	// ProjectBoardList is a list of all project boards in a repository
+	ProjectBoardList []*ProjectBoard
+)
+
+const (
+	// ProjectBoardTypeNone is a project board type that has no predefined columns
+	ProjectBoardTypeNone ProjectBoardType = iota
+
+	// ProjectBoardTypeBasicKanban is a project board type that has basic predefined columns
+	ProjectBoardTypeBasicKanban
+
+	// ProjectBoardTypeBugTriage is a project board type that has predefined columns suited to hunting down bugs
+	ProjectBoardTypeBugTriage
+)
+
+// ProjectBoard is used to represent boards on a project
+type ProjectBoard struct {
+	ID      int64 `xorm:"pk autoincr"`
+	Title   string
+	Default bool `xorm:"NOT NULL DEFAULT false"` //if true it collects issues witch are not signed to a specific board jet
```

Comment:
Since we have a memory board `Uncategory`, it seems the column `default` is unused?

Reply:
this is for the default Board witch we would add in another pull - so we dont need a new migration - through I can remove it

Reply:
I vote remove. Another migration is not a bad thing.

Reply:
I originally said to remove but now I see how this that would work.  I think this should stay actually.

Reply:
Actually thinking on the correct place for this is on the project - it should have a default project_board_id instead.